src/pagination/dates.rs

```diff
@@ -0,0 +1,159 @@
+        parent_dates
+    } else {
+        vec![]
+    };
+    if let Some(_field) = date_holder.field {
+        sort_posts(&mut date_holder.posts, &config);
+        current_date.push(date_holder.value);
+        let index_title = liquid::value::Value::array(date_fields_to_array(&current_date));
+        let cur_date_paginators =
+            create_all_paginators(&date_holder.posts, &doc, &config, Some(&index_title))?;
+        if !cur_date_paginators.is_empty() {
+            cur_date_holder_paginators.extend(cur_date_paginators.into_iter());
+        } else {
+            let mut p = Paginator::default();
+            p.index_title = Some(index_title);
+            cur_date_holder_paginators.push(p);
+        }
+    } else {
+        cur_date_holder_paginators.push(Paginator::default());
+    }
+    for mut dh in &mut date_holder.sub_date {
+        let mut sub_paginators_holder =
+            walk_dates(&mut dh, &config, &doc, Some(current_date.clone()))?;
+
+        if let Some(indexes) = cur_date_holder_paginators[0].indexes.as_mut() {
+            indexes.push(sub_paginators_holder[0].clone());
+        } else {
+            cur_date_holder_paginators[0].indexes = Some(vec![sub_paginators_holder[0].clone()]);
+        }
+        cur_date_holder_paginators.append(&mut sub_paginators_holder);
+    }
+    Ok(cur_date_holder_paginators)
+}
+
+fn find_or_create_date_holder_and_put_post<'a, 'b>(
+    date_holder: &'b mut DateIndexHolder<'a>,
+    published_date: &DateTime,
+    wanted_field: &DateIndex,
+    post: &'a liquid::value::Value,
+) {
+    let value = get_date_field_value(&published_date, &wanted_field);
+    let mut not_found = true;
+    for mut dh in date_holder.sub_date.iter_mut() {
+        let dh_field = dh
+            .field
+            .expect("Only root has None, we should always have a field");
+        if dh_field < *wanted_field {
+            // not at the level we want but still need to find the correct parent
+            // parent should have been created in a previous loop
+            let parent_value = get_date_field_value(&published_date, &dh_field);
+            if dh.value == parent_value {
+                find_or_create_date_holder_and_put_post(
+                    &mut dh,
+                    &published_date,
+                    wanted_field,
+                    &post,
+                );
+                not_found = false;
+            }
+        } else if dh_field == *wanted_field && dh.value == value {
+            dh.posts.push(post);
+            not_found = false;
+        }
+    }
+    // not found create one
+    if not_found {
+        let mut holder = DateIndexHolder::new(value, Some(*wanted_field));
+        holder.posts.push(post);
+        date_holder.sub_date.push(holder);
+    }
+}
+
+fn get_date_field_value(date: &DateTime, field: &DateIndex) -> u32 {
+    match field {
+        DateIndex::Year => date.year() as u32, // let's say we don't support negative year…
```

Comment:
Should we at least turn this comment into a panic?

Reply:
You're totally right about this, going to do that